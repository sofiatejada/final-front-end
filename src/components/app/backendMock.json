[
  {
      "id": "8",
      "name": "Nancy",
      "image": "https://vignette.wikia.nocookie.net/heyarnold/images/6/62/Nancy.jpg/revision/latest/scale-to-width-down/310?cb=20110423055923"
  },
  {
      "id": "9",
      "name": "Phoebe Heyerdahl",
      "image": "https://vignette.wikia.nocookie.net/heyarnold/images/9/93/Phoebe_%28classic%29.png/revision/latest/scale-to-width-down/247?cb=20190816022206"
  },
  {
      "id": "10",
      "name": "Timberly Johanssen",
      "image": "https://vignette.wikia.nocookie.net/heyarnold/images/e/e8/Timberly.png/revision/latest/scale-to-width-down/310?cb=20171210045406"
  }
]
