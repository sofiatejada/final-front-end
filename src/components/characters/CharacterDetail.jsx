import React from 'react';
import { Link } from 'react-router-dom';
import useCharacterDetail from '../../hooks/useHeyArnoldCharacterDetail';

const backendURL = 'https://full-stack-be.herokuapp.com/api/v1/favorites';
// eslint-disable-next-line max-len
const spinner = 'https://images-wixmp-ed30a86b8c4ca887773594c2.wixmp.com/f/945673bd-feaf-46f2-9e69-82deb505b95a/d9uqsyv-367865ee-f6f7-48ed-ba74-bbfa2a36a52a.gif?token=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJzdWIiOiJ1cm46YXBwOjdlMGQxODg5ODIyNjQzNzNhNWYwZDQxNWVhMGQyNmUwIiwiaXNzIjoidXJuOmFwcDo3ZTBkMTg4OTgyMjY0MzczYTVmMGQ0MTVlYTBkMjZlMCIsIm9iaiI6W1t7InBhdGgiOiJcL2ZcLzk0NTY3M2JkLWZlYWYtNDZmMi05ZTY5LTgyZGViNTA1Yjk1YVwvZDl1cXN5di0zNjc4NjVlZS1mNmY3LTQ4ZWQtYmE3NC1iYmZhMmEzNmE1MmEuZ2lmIn1dXSwiYXVkIjpbInVybjpzZXJ2aWNlOmZpbGUuZG93bmxvYWQiXX0.N-qjErA2OUW5Hz32FymO9LfJUBrnvqgY378NpbRSmbo';

export default function CharacterDetail() {

  const { character, loading } = useCharacterDetail();

  if(loading) {
    return (
      <img src={spinner} alt="loading gif" />
    );
  }

  const saveCharacter = async (character) => {
    fetch(`${backendURL}`, {
      method: 'POST',
      headers: {
        'Content-Type': 'application/json'
      },
      body: JSON.stringify(character)
    })
      .then((sentData) => sentData.json());
  };

  const handleSubmit = (event) => {
    event.preventDefault();
    saveCharacter(character);
    // .then(window.location = '/personal/faves');
  };

  return (
    <main>
      <header>
        <Link to="/">
          Homepage
        </Link>
        <Link to="/personal/faves">
          Favorites
        </Link>
      </header>
      <figure>
        <img src={character.image} alt={`image of ${character.name}`} />
        <figcaption>
          {character.name}
        </figcaption>
      </figure>
      <form onSubmit={handleSubmit}>
        <button aria-label="add-butt" type="submit">Add to Faves</button>
      </form>
    </main>
  );
}
