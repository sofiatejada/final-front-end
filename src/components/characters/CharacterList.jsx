import React from 'react';
import { Link } from 'react-router-dom';
import Character from './Character';
import useCharacters from '../../hooks/useHeyArnoldCharacters';

// eslint-disable-next-line max-len
const spinner = 'https://images-wixmp-ed30a86b8c4ca887773594c2.wixmp.com/f/945673bd-feaf-46f2-9e69-82deb505b95a/d9uqsyv-367865ee-f6f7-48ed-ba74-bbfa2a36a52a.gif?token=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJzdWIiOiJ1cm46YXBwOjdlMGQxODg5ODIyNjQzNzNhNWYwZDQxNWVhMGQyNmUwIiwiaXNzIjoidXJuOmFwcDo3ZTBkMTg4OTgyMjY0MzczYTVmMGQ0MTVlYTBkMjZlMCIsIm9iaiI6W1t7InBhdGgiOiJcL2ZcLzk0NTY3M2JkLWZlYWYtNDZmMi05ZTY5LTgyZGViNTA1Yjk1YVwvZDl1cXN5di0zNjc4NjVlZS1mNmY3LTQ4ZWQtYmE3NC1iYmZhMmEzNmE1MmEuZ2lmIn1dXSwiYXVkIjpbInVybjpzZXJ2aWNlOmZpbGUuZG93bmxvYWQiXX0.N-qjErA2OUW5Hz32FymO9LfJUBrnvqgY378NpbRSmbo';

export default function CharacterList() {

  const { characterList, loading } = useCharacters();

  if(loading) {
    return (
      <img src={spinner} alt="loading gif" />
    );
  }
 
  return (
    <>
      <header>
        <Link to="/">
          Homepage
        </Link>
        <Link to="/personal/faves">
          Favorites
        </Link>
      </header>
      <ul aria-label="character-list" role="list">
        {characterList.map((character) => (
          <li key={character.id}>
            <Link key={character.id} to={`/character/${character.id}`}>
              <Character 
                id={character.id} 
                name={character.name} 
                image={character.image} />
            </Link>
          </li>
        ))}
      </ul>
    </>
  );
}
