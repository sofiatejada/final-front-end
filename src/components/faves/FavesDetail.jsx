import React, { useState, useEffect } from 'react';
import { Link } from 'react-router-dom';
import { useParams } from 'react-router-dom';

const backendURL = 'https://full-stack-be.herokuapp.com/api/v1/favorites';
// eslint-disable-next-line max-len
const spinner = 'https://images-wixmp-ed30a86b8c4ca887773594c2.wixmp.com/f/945673bd-feaf-46f2-9e69-82deb505b95a/d9uqsyv-367865ee-f6f7-48ed-ba74-bbfa2a36a52a.gif?token=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJzdWIiOiJ1cm46YXBwOjdlMGQxODg5ODIyNjQzNzNhNWYwZDQxNWVhMGQyNmUwIiwiaXNzIjoidXJuOmFwcDo3ZTBkMTg4OTgyMjY0MzczYTVmMGQ0MTVlYTBkMjZlMCIsIm9iaiI6W1t7InBhdGgiOiJcL2ZcLzk0NTY3M2JkLWZlYWYtNDZmMi05ZTY5LTgyZGViNTA1Yjk1YVwvZDl1cXN5di0zNjc4NjVlZS1mNmY3LTQ4ZWQtYmE3NC1iYmZhMmEzNmE1MmEuZ2lmIn1dXSwiYXVkIjpbInVybjpzZXJ2aWNlOmZpbGUuZG93bmxvYWQiXX0.N-qjErA2OUW5Hz32FymO9LfJUBrnvqgY378NpbRSmbo';

export default function FavesDetail() {

  const [character, setCharacter] = useState();
  const [loading, setLoading] = useState(true);
  const [changedName, setNameChange] = useState(null);
  const [changedImage, setImageChange] = useState(null);
  const { id } = useParams();

  const fetchBackendData = async () => {
    const res = await fetch(`${backendURL}/${id}`);
    const data = await res.json();
    console.log(data);
    return data;
  };

  useEffect(() => {
    fetchBackendData()
      .then((character) => setCharacter(character))
      .finally(() => setLoading(false));
  }, []);

  const removeCharacter = async (id) => {
    const res = await fetch(`${backendURL}/${id}`, {
      method: 'DELETE'
    });
    const deleted = await res.json();
      
    return deleted;
  };

  const updateCharacter = async (id) => {

    const res = await fetch(`${backendURL}/${id}`, {
      method: 'PUT',
      headers: {
        'Content-Type': 'application/json'
      },
      body: JSON.stringify({
        name: changedName ?? character.name,
        image: changedImage ?? character.image,
      })
    });
    const formData = await res.json();
    console.log(formData);
      
    return formData;
  };

  const handleDeleteSubmit = (event) => {
    event.preventDefault();
    removeCharacter(id);
    // .then(window.location = '/');
  };

  const handlePutSubmit = (event) => {
    event.preventDefault();
    updateCharacter(id);
  }

  const handleNameChange = (event) => {
    setNameChange(event.target.value)
  }
  const handleImageChange = (event) => {
    setImageChange(event.target.value)
  }

  if(loading) {
    return (
      <img src={spinner} alt="loading gif" />
    );
  }

  return (
    <main>
      <header>
        <Link to="/">
          Homepage
        </Link>
        <Link to="/personal/faves">
          Favorites
        </Link>
      </header>
      <figure>
        <img src={character.image} alt={`image of ${character.name}`} />
        <figcaption>
          {character.name}
        </figcaption>
      </figure>
      <form onSubmit={handleDeleteSubmit}>
        <button>Remove from Faves</button>
      </form>
      <form onSubmit={handlePutSubmit}>
        <input onChange={handleNameChange} type="text" placeholder="name" />
        <input onChange={handleImageChange} type="url" placeholder="image" />
        <button>Change stuff!</button>
      </form>
    </main>
  );
}
