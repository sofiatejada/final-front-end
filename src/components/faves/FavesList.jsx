import React from 'react';
import { useEffect } from 'react';
import { useState } from 'react';
import Fave from './Fave';
import { Link } from 'react-router-dom';

const backendURL = 'https://full-stack-be.herokuapp.com/api/v1/favorites';
// eslint-disable-next-line max-len
const spinner = 'https://images-wixmp-ed30a86b8c4ca887773594c2.wixmp.com/f/945673bd-feaf-46f2-9e69-82deb505b95a/d9uqsyv-367865ee-f6f7-48ed-ba74-bbfa2a36a52a.gif?token=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJzdWIiOiJ1cm46YXBwOjdlMGQxODg5ODIyNjQzNzNhNWYwZDQxNWVhMGQyNmUwIiwiaXNzIjoidXJuOmFwcDo3ZTBkMTg4OTgyMjY0MzczYTVmMGQ0MTVlYTBkMjZlMCIsIm9iaiI6W1t7InBhdGgiOiJcL2ZcLzk0NTY3M2JkLWZlYWYtNDZmMi05ZTY5LTgyZGViNTA1Yjk1YVwvZDl1cXN5di0zNjc4NjVlZS1mNmY3LTQ4ZWQtYmE3NC1iYmZhMmEzNmE1MmEuZ2lmIn1dXSwiYXVkIjpbInVybjpzZXJ2aWNlOmZpbGUuZG93bmxvYWQiXX0.N-qjErA2OUW5Hz32FymO9LfJUBrnvqgY378NpbRSmbo';

export default function FavesList() {
  const [backendList, setBackendList] = useState([]);
  const [loading, setLoading] = useState(true);
  
  const fetchBackendData = async () => {
    const res = await fetch(`${backendURL}`);
    const data = await res.json();

    return data;
  };

  useEffect(() => {
    fetchBackendData()
      .then((list) => setBackendList(list))
      .finally(() => setLoading(false));
  }, []);

  if(loading) {
    return (
      <img src={spinner} alt="loading gif" />
    );
  }
 
  return (
    <>
      <header>
        <Link to="/">
          Homepage
        </Link>
        <Link to="/personal/faves">
          Favorites
        </Link>
      </header>
      <ul aria-label="fave-list" role="list">
        {backendList.map((character) => (
          <li key={character.id}>
            <Fave 
              id={character.id} name={character.name} image={character.image} />
          </li>
        ))}
      </ul>
    </>
  );
}
