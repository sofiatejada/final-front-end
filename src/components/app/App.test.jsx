import React from 'react';
import { render, screen, fireEvent, waitFor } from '@testing-library/react';
import CharacterList from '../characters/CharacterList';
import Character from '../characters/Character';
import { MemoryRouter } from 'react-router-dom';
import { rest } from 'msw';
import { setupServer } from 'msw/node';
import mock from './mock.json';
import backendMock from './backendMock.json';
import FavesList from '../faves/FavesList';
import FavesDetail from '../faves/FavesDetail';
import CharacterDetail from '../characters/CharacterDetail';

const server = setupServer(
  // eslint-disable-next-line max-len
  rest.get('https://hey-arnold-api.herokuapp.com/api/v1/characters/', 
    (req, res, ctx) => {
      return res(
        ctx.json(mock)
      );
    })
);

const beServer = setupServer(
  // eslint-disable-next-line max-len
  rest.get('https://full-stack-be.herokuapp.com/api/v1/favorites', 
    (req, res, ctx) => {
      return res(
        ctx.json(backendMock)
      );
    })
);

const philServer = setupServer(
  // eslint-disable-next-line max-len
  rest.get('https://hey-arnold-api.herokuapp.com/api/v1/characters/:id', 
    (req, res, ctx) => {
      return res(
        ctx.json({
          "_id": "5da237699734fdcb7bef8f52",
          "name": "Grandpa Phil",
          "image": "https://vignette.wikia.nocookie.net/heyarnold/images/7/79/GrandpaPhil.png/revision/latest/scale-to-width-down/300?cb=20160924030436"
      },)
      );
    })
);

const nancyServer = setupServer(
  // eslint-disable-next-line max-len
  rest.get('https://full-stack-be.herokuapp.com/api/v1/favorites/:id', 
    (req, res, ctx) => {
      return res(
        ctx.json({
          "id": "8",
          "name": "Nancy",
          "image": "https://vignette.wikia.nocookie.net/heyarnold/images/6/62/Nancy.jpg/revision/latest/scale-to-width-down/310?cb=20110423055923"
      },)
      );
    })
);

describe('character component', () => {

  it('displays a character', () => {
    render(
      <Character 
        id="1" 
        name="poop"
        image="example.png" />
    );

    const article = screen.getByRole('img', { name: 'image of poop' });
    expect(article).toMatchSnapshot();
  });
});

describe('character list component', () => {

  beforeAll(() => server.listen());
  afterAll(() => server.close());

  it('displays a list of characters', async () => {
    render(
      <MemoryRouter>
        <CharacterList />
      </MemoryRouter>
    );
    
    screen.getByAltText('loading gif');
    const list = await screen.findByLabelText('character-list');
    screen.debug();
    expect(list).toMatchSnapshot();
  });
});

describe('faves list', () => {

  beforeAll(() => beServer.listen());
  afterAll(() => beServer.close());

  it('displays list of faves', async () => {
    render(
      <MemoryRouter>
        <FavesList />
      </MemoryRouter>
    );

    screen.getByAltText('loading gif');
    const list = await screen.findByLabelText('fave-list');
    screen.debug();
    expect(list).toMatchSnapshot();
  });

});

describe('adding phil', () => {

  beforeAll(() => philServer.listen());
  afterAll(() => philServer.close());

  it('adds phil', async () => {
    render(
      <MemoryRouter initialEntries={['/5da237699734fdcb7bef8f52']} >
        <CharacterDetail />
        <FavesList />
      </MemoryRouter>
    );
  
    const phil = await screen.findByAltText('image of Grandpa Phil');
    fireEvent.click(phil);
  
    const addbutt = await screen.findByLabelText('add-butt');
    fireEvent.click(addbutt);
  
    screen.getByAltText('loading gif');
    const favePhil = await screen.findByAltText('fave image of Grandpa Phil')
    waitFor(() => {
      expect(favePhil).not.toBeNull();
    })
  });
});

describe('removing nancy', () => {

  beforeAll(() => nancyServer.listen());
  afterAll(() => nancyServer.close());

  it('removes nancy', async () => {
    render(
      <MemoryRouter initialEntries={['/8']} >
        <FavesDetail />
        <FavesList />
      </MemoryRouter>
    );
  
    const nancy = await screen.findByAltText('fave image of Nancy');
    fireEvent.click(nancy);
  
    const removebutt = await screen.findByLabelText('remove-butt');
    fireEvent.click(removebutt);
  
    screen.getByAltText('loading gif');
    waitFor(() => {
      expect(nancy).toBeNull();
    })
  });
});
