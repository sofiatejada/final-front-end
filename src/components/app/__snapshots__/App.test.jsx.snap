// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`character component displays a character 1`] = `
<img
  alt="image of poop"
  src="example.png"
/>
`;

exports[`character list component displays a list of characters 1`] = `
<ul
  aria-label="character-list"
  role="list"
>
  <li>
    <a
      href="/character/5da237699734fdcb7bef8f5c"
    >
      <figure>
        <img
          alt="image of Nancy"
          src="https://vignette.wikia.nocookie.net/heyarnold/images/6/62/Nancy.jpg/revision/latest/scale-to-width-down/310?cb=20110423055923"
        />
        <figcaption>
          Nancy
        </figcaption>
      </figure>
    </a>
  </li>
  <li>
    <a
      href="/character/5da237699734fdcb7bef8f59"
    >
      <figure>
        <img
          alt="image of Grandpa's grandfather"
          src="https://vignette.wikia.nocookie.net/heyarnold/images/f/f7/Grandpa%27s_grandfather.png/revision/latest/scale-to-width-down/310?cb=20190228081015"
        />
        <figcaption>
          Grandpa's grandfather
        </figcaption>
      </figure>
    </a>
  </li>
  <li>
    <a
      href="/character/5da237699734fdcb7bef8f52"
    >
      <figure>
        <img
          alt="image of Grandpa Phil"
          src="https://vignette.wikia.nocookie.net/heyarnold/images/7/79/GrandpaPhil.png/revision/latest/scale-to-width-down/300?cb=20160924030436"
        />
        <figcaption>
          Grandpa Phil
        </figcaption>
      </figure>
    </a>
  </li>
  <li>
    <a
      href="/character/5da237699734fdcb7bef8f58"
    >
      <figure>
        <img
          alt="image of Grandpa's father"
          src="https://vignette.wikia.nocookie.net/heyarnold/images/e/e0/Grandpa%27s_father.png/revision/latest/scale-to-width-down/310?cb=20190228081112"
        />
        <figcaption>
          Grandpa's father
        </figcaption>
      </figure>
    </a>
  </li>
  <li>
    <a
      href="/character/5da237699734fdcb7bef8f70"
    >
      <figure>
        <img
          alt="image of Phoebe Heyerdahl"
          src="https://vignette.wikia.nocookie.net/heyarnold/images/9/93/Phoebe_%28classic%29.png/revision/latest/scale-to-width-down/247?cb=20190816022206"
        />
        <figcaption>
          Phoebe Heyerdahl
        </figcaption>
      </figure>
    </a>
  </li>
  <li>
    <a
      href="/character/5da237699734fdcb7bef8f7e"
    >
      <figure>
        <img
          alt="image of Eugene Horowitz"
          src="https://vignette.wikia.nocookie.net/heyarnold/images/9/99/EAFDB121-EC13-4F12-849E-37037A95E3D6.png/revision/latest/scale-to-width-down/270?cb=20190816022130"
        />
        <figcaption>
          Eugene Horowitz
        </figcaption>
      </figure>
    </a>
  </li>
  <li>
    <a
      href="/character/5da237699734fdcb7bef8f82"
    >
      <figure>
        <img
          alt="image of Ray"
          src="https://vignette.wikia.nocookie.net/heyarnold/images/3/34/Ray.png/revision/latest/scale-to-width-down/310?cb=20181006091830"
        />
        <figcaption>
          Ray
        </figcaption>
      </figure>
    </a>
  </li>
  <li>
    <a
      href="/character/5da237699734fdcb7bef8f8b"
    >
      <figure>
        <img
          alt="image of Shelley"
          src="https://vignette.wikia.nocookie.net/heyarnold/images/c/cb/Shelley.jpg/revision/latest/scale-to-width-down/310?cb=20120527025256"
        />
        <figcaption>
          Shelley
        </figcaption>
      </figure>
    </a>
  </li>
  <li>
    <a
      href="/character/5da237699734fdcb7bef8f76"
    >
      <figure>
        <img
          alt="image of Lila Sawyer"
          src="https://vignette.wikia.nocookie.net/heyarnold/images/b/b0/Lila_Sawyer.png/revision/latest/scale-to-width-down/310?cb=20171212053404"
        />
        <figcaption>
          Lila Sawyer
        </figcaption>
      </figure>
    </a>
  </li>
  <li>
    <a
      href="/character/5da237699734fdcb7bef8f7c"
    >
      <figure>
        <img
          alt="image of Uncle Stinky"
          src="https://vignette.wikia.nocookie.net/heyarnold/images/8/89/Uncle_Stinky.jpg/revision/latest/scale-to-width-down/310?cb=20120924204406"
        />
        <figcaption>
          Uncle Stinky
        </figcaption>
      </figure>
    </a>
  </li>
  <li>
    <a
      href="/character/5da237699734fdcb7bef8f55"
    >
      <figure>
        <img
          alt="image of Arnie"
          src="https://vignette.wikia.nocookie.net/heyarnold/images/4/42/Arnie.jpg/revision/latest/scale-to-width-down/310?cb=20110109195825"
        />
        <figcaption>
          Arnie
        </figcaption>
      </figure>
    </a>
  </li>
  <li>
    <a
      href="/character/5da237699734fdcb7bef8f66"
    >
      <figure>
        <img
          alt="image of Olga Pataki"
          src="https://vignette.wikia.nocookie.net/heyarnold/images/5/5e/Olga_Pataki.png/revision/latest/scale-to-width-down/310?cb=20180124053913"
        />
        <figcaption>
          Olga Pataki
        </figcaption>
      </figure>
    </a>
  </li>
  <li>
    <a
      href="/character/5da237699734fdcb7bef8f6b"
    >
      <figure>
        <img
          alt="image of Timberly Johanssen"
          src="https://vignette.wikia.nocookie.net/heyarnold/images/e/e8/Timberly.png/revision/latest/scale-to-width-down/310?cb=20171210045406"
        />
        <figcaption>
          Timberly Johanssen
        </figcaption>
      </figure>
    </a>
  </li>
  <li>
    <a
      href="/character/5da237699734fdcb7bef8f6d"
    >
      <figure>
        <img
          alt="image of Jerry Berman"
          src="https://vignette.wikia.nocookie.net/heyarnold/images/2/2f/Jerry_Berman.png/revision/latest/scale-to-width-down/310?cb=20171219061306"
        />
        <figcaption>
          Jerry Berman
        </figcaption>
      </figure>
    </a>
  </li>
  <li>
    <a
      href="/character/5da237699734fdcb7bef8f73"
    >
      <figure>
        <img
          alt="image of Rhonda Wellington Lloyd"
          src="https://vignette.wikia.nocookie.net/heyarnold/images/b/b4/C1BBB448-FE38-4F7B-98FA-2137F277F9C5.png/revision/latest/scale-to-width-down/213?cb=20190816022649"
        />
        <figcaption>
          Rhonda Wellington Lloyd
        </figcaption>
      </figure>
    </a>
  </li>
  <li>
    <a
      href="/character/5da237699734fdcb7bef8f78"
    >
      <figure>
        <img
          alt="image of Stinky Peterson"
          src="https://vignette.wikia.nocookie.net/heyarnold/images/2/24/Stinky_Peterson.png/revision/latest/scale-to-width-down/310?cb=20160516072820"
        />
        <figcaption>
          Stinky Peterson
        </figcaption>
      </figure>
    </a>
  </li>
  <li>
    <a
      href="/character/5da237699734fdcb7bef8f71"
    >
      <figure>
        <img
          alt="image of Kyo Heyerdahl"
          src="https://vignette.wikia.nocookie.net/heyarnold/images/c/cf/Kyo_Heyerdahl.png/revision/latest/scale-to-width-down/310?cb=20190310221807"
        />
        <figcaption>
          Kyo Heyerdahl
        </figcaption>
      </figure>
    </a>
  </li>
  <li>
    <a
      href="/character/5da237699734fdcb7bef8f8e"
    >
      <figure>
        <img
          alt="image of Iggy"
          src="https://vignette.wikia.nocookie.net/heyarnold/images/0/0e/Iggy.png/revision/latest/scale-to-width-down/310?cb=20190210045957"
        />
        <figcaption>
          Iggy
        </figcaption>
      </figure>
    </a>
  </li>
  <li>
    <a
      href="/character/5da237699734fdcb7bef8f77"
    >
      <figure>
        <img
          alt="image of Mr. Sawyer"
          src="https://vignette.wikia.nocookie.net/heyarnold/images/4/45/Mr._Sawyer.png/revision/latest/scale-to-width-down/310?cb=20171219055204"
        />
        <figcaption>
          Mr. Sawyer
        </figcaption>
      </figure>
    </a>
  </li>
  <li>
    <a
      href="/character/5da237699734fdcb7bef8f84"
    >
      <figure>
        <img
          alt="image of Thaddeus \\"Curly\\" Gammelthorpe"
          src="https://vignette.wikia.nocookie.net/heyarnold/images/b/ba/Classic_Curly_promo.png/revision/latest/scale-to-width-down/246?cb=20181010011614"
        />
        <figcaption>
          Thaddeus "Curly" Gammelthorpe
        </figcaption>
      </figure>
    </a>
  </li>
</ul>
`;

exports[`faves list displays list of faves 1`] = `
<ul
  aria-label="fave-list"
  role="list"
>
  <li>
    <a
      href="/personal/faves/8"
    >
      <figure>
        <img
          alt="fave image of Nancy"
          src="https://vignette.wikia.nocookie.net/heyarnold/images/6/62/Nancy.jpg/revision/latest/scale-to-width-down/310?cb=20110423055923"
        />
        <figcaption>
          Nancy
        </figcaption>
      </figure>
    </a>
  </li>
  <li>
    <a
      href="/personal/faves/9"
    >
      <figure>
        <img
          alt="fave image of Phoebe Heyerdahl"
          src="https://vignette.wikia.nocookie.net/heyarnold/images/9/93/Phoebe_%28classic%29.png/revision/latest/scale-to-width-down/247?cb=20190816022206"
        />
        <figcaption>
          Phoebe Heyerdahl
        </figcaption>
      </figure>
    </a>
  </li>
  <li>
    <a
      href="/personal/faves/10"
    >
      <figure>
        <img
          alt="fave image of Timberly Johanssen"
          src="https://vignette.wikia.nocookie.net/heyarnold/images/e/e8/Timberly.png/revision/latest/scale-to-width-down/310?cb=20171210045406"
        />
        <figcaption>
          Timberly Johanssen
        </figcaption>
      </figure>
    </a>
  </li>
</ul>
`;
